lib/vsc/utils/exceptions.py

```diff
@@ -81,6 +84,30 @@ def __init__(self, msg, *args, **kwargs):
         if args:
             msg = msg % args
 
+        if self.LOC_INFO_TOP_PKG_NAMES is not None:
+            # determine correct frame to fetch location information from
+            frames_up = 1
+            if self.__class__ != LoggedException:
+                # move a level up when this instance is derived from LoggedException
+                frames_up += 1
+
+            # figure out where error was raised from
+            # current frame: this constructor, one frame above: location where this LoggedException was created/raised
+            frameinfo = inspect.getouterframes(inspect.currentframe())[frames_up]
+
+            # determine short location of Python module where error was raised from,
+            # i.e. starting with an entry from LOC_INFO_TOP_PKG_NAMES
+            path_parts = frameinfo[1].split(os.path.sep)
+            top_indices = [path_parts.index(n) for n in self.LOC_INFO_TOP_PKG_NAMES if n in path_parts]
+            if top_indices:
```

Comment:
replace if/else with
```python
relpath = os.path.join(*path_parts[max(top_indices or [0]):])
```
(or `+` instead of `or`; it doesn't really matter)